components/tikv_alloc/src/lib.rs

```diff
@@ -0,0 +1,290 @@
+//! platforms override C malloc, including macOS_. This means on some
+//! platforms RocksDB is using the system malloc. On Linux C malloc is
+//! redirected to jemalloc.
+//!
+//! This crate accepts two cargo features:
+//!
+//! - mem-profiling - compiles jemalloc and this crate with profiling
+//!   capability
+//!
+//! - no-jemalloc - compiles without jemalloc, such that tikv
+//!   will use the system allocator
+//!
+//! cfg `fuzzing` is defined by `run_libfuzzer` in `fuzz/cli.rs` and
+//! is passed to rustc directly with `--cfg`; in other words it's not
+//! controlled through a crate feature.
+//!
+//! Ideally there should be no jemalloc-specific code outside this
+//! crate.
+//!
+//! # Profiling
+//!
+//! Profiling with jemalloc requires both build-time and run-time
+//! configuration. At build time cargo needs the `--mem-profiling`
+//! feature, and at run-time jemalloc needs to set the `opt.prof`
+//! option to true, ala `MALLOC_CONF="opt.prof:true".
+//!
+//! In production you might also set `opt.prof_active` to `false` to
+//! keep profiling off until there's an incident. Jemalloc has
+//! a variety of run-time [profiling options].
+//!
+//! [profiling options]: http://jemalloc.net/jemalloc.3.html#opt.prof
+//!
+//! Here's an example of how you might build and run via cargo, with
+//! profiling:
+//!
+//! ```notrust
+//! export MALLOC_CONF="prof:true,prof_active:false,prof_prefix:$(pwd)/jeprof"
+//! cargo test --features mem-profiling -p tikv_alloc -- --ignored
+//! ```
+//!
+//! (In practice you might write this as a single statement, setting
+//! `MALLOC_CONF` only temporarily, e.g. `MALLOC_CONF="..." cargo test
+//! ...`).
+//!
+//! When running cargo while `prof:true`, you will see messages like
+//!
+//! ```notrust
+//! <jemalloc>: Invalid conf pair: prof:true
+//! <jemalloc>: Invalid conf pair: prof_active:false
+//! ```
+//!
+//! This is normal - they are being emitting by the jemalloc in cargo
+//! and rustc, which are both configured without profiling. tikv's
+//! jemalloc is configured for profiling if you pass
+//! `--features=mem-profiling` to cargo for eather `tikv_alloc` or
+//! `tikv`.
+
+#[cfg(all(unix, not(fuzzing), not(feature = "no-jemalloc")))]
+extern crate jemallocator;
+extern crate libc;
+#[cfg(feature = "mem-profiling")]
+#[macro_use]
+extern crate log;
+
+// The global allocator, usually.
+#[cfg(all(unix, not(fuzzing), not(feature = "no-jemalloc")))]
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
+
+// The global allocator, without jemalloc.
+// This is only necessary prior to the removal of jemalloc from
+// Rust. Afterwards this might be removed.
+#[cfg(not(all(unix, not(fuzzing), not(feature = "no-jemalloc"))))]
+#[global_allocator]
+static ALLOC: std::alloc::System = std::alloc::System;
+
+pub use self::impl_::*;
+
+// The implementation of this crate when jemalloc is turned on
+#[cfg(all(unix, not(fuzzing), not(feature = "no-jemalloc")))]
+mod impl_ {
```

Comment:
em, is it common to add a `_` as the prefix of the mod? Seem we don't use this style in other places in TiKV.